### ISSUE TITLE
PulsarApi.proto

### ISSUE THREAD

**Issue by vilinski** [ctaggart/froto#106]
Getting error with the file [PulsarApi.proto](https://github.com/apache/pulsar/blob/master/pulsar-common/src/main/proto/PulsarApi.proto):

```
falanx --inputfile PulsarApi.proto --outputfile Pulsar.Proto.fs --defaultnamespace Pulsar.Proto --serializer binary
Generating code for: PulsarApi.proto
ERROR: proto file was not able to be parsed.

Error in Ln: 55 Col: 11
    optional int32 partition = 3;
          ^
Expecting: whitespace
```
Looks nothing suspect for me on this place:

```
message MessageIdData {
    required uint64 ledgerId = 1;
    required uint64 entryId  = 2;
    optional int32 partition = 3 [default = -1];
    optional int32 batch_index = 4 [default = -1];
}
```

**Comment by jhugard** [comment]
Haven't worked this code in forever, but I think the root cause has to do with the core parsing for messages: something was consuming a token and leaving the parser in a bad state.

Can you try this patch in Parser.fs?

```fsharp
        // Message statement: field | enum | etc.
        and internal pMessageStatement =
            let pMessageMessage =
                pMessageDef |>> TMessageMessage

            choice [
                (isProto2 >>. (attempt <| pGroup))
                (isProto2 >>. (attempt <| pMessageExtend))
                (isProto2 >>. (attempt <| pExtensions))
                (attempt <| pMessageOption)
                (attempt <| pMap)
                (attempt <| pReserved)
                (attempt <| pOneOf)
                (attempt <| pMessageEnum)
                (attempt <| pMessageMessage)
                pField                 // must be parsed last for keywords to not be considered type names 

                ]
```

**Comment by jhugard** [comment]
@vilinski The commit above was included in release 0.7.2.  Can you verify the change worked, so that this issue can be closed?

**Comment by vilinski** [comment]
sorry, waited for falanx updates it's deps. 

Falanx 0.6.0 still doesn't work with actual `PulsarApi.proto` with the parser bug on the same line.

```
❯ falanx --inputfile PulsarApi.proto --outputfile Pulsar.Proto.fs --defaultnamespace Pulsar.Proto --serializer binary

Generating code for: PulsarApi.proto
ERROR: proto file was not able to be parsed.

Error in Ln: 60 Col: 12
     optional int32 partition = 3 [default = -1];
           ^
Expecting: whitespace
```

But type provider works without errors.

```fsharp
#r "nuget: Froto.TypeProvider"

open Froto.TypeProvider

let [<Literal>] PulsarApiProto = __SOURCE_DIRECTORY__ + "/PulsarApi.proto"
type PulsarApi = ProtocolBuffersTypeProvider<PulsarApiProto>

let msg = PulsarApi.Pulsar.Proto.MessageIdData()
msg.Partition <- Some 42
```

so not a Froto issue anymore.
Rather falanx still does not include the changes (09.04.2020)
